src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs

```diff
@@ -1172,6 +1172,42 @@ private static string AddToPath(string basePath, string pathToAdd, int insertPos
             return result.ToString();
         }
 
+        /// <summary>
+        /// The available module path types.
+        /// </summary>
+        public enum PSModulePathType
+        {
+            /// <summary>The users module path.</summary>
+            User,
+
+            /// <summary>The PSHOME module path.</summary>
+            PSHome,
+
+            /// <summary>The shared module path.</summary>
+            Shared
+        }
+
+        /// <summary>
+        /// Retrieve the current PSModulePath of the specified type.
+        /// </summary>
+        /// <param name="type">The type of module path to retrieve. This can be User, PSHome, or Shared.</param>
+        /// <returns>The string representing the requested module path type.</returns>
+        public static string GetPSModulePath(PSModulePathType type)
```

Comment:
I am wondering if the parameter name should be `scope` to be consistent with other cmdlets and PowerShellGet.